### PR TITLE
Added Data61 Elastic Search GNAF Server as Search Provider

### LINKED ISSUE #1588: Add GNAF support to search

### PR DESCRIPTION
**Pull request by AlexGilleran**
Fixes #1588

### CODE REVIEW

File: lib/Models/Cesium.js [hunk 1]
```diff
@@ -757,6 +757,35 @@ Cesium.prototype._buildPickedFeatures = function(providerCoords, pickPosition, e
     return result;
 };
 
+Cesium.prototype.getBoundingBoxCoords = function() {
```
Comment (kring) [inline lib/Models/Cesium.js]: Doesn't this do the same thing as `getCurrentExtent`, except this one won't work if the globe isn't in the corners of the screen? :)

File: lib/Models/GNAFApi.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+var DEFAULT_MAX_RESULTS = 10;
+var MIN_SCORE = 1;
+var DATA61_GNAF_SEARCH_URL = 'http://gnaf.it.csiro.au/es/_search';
+
+/**
+ * Simple JS Api for the Data61 ElasticSearch GNAF service.
+ *
+ * @param corsProxy CorsProxy to use to determine whether calls need to go through the terria proxy.
+ * @param [overrideUrl] The URL to use to query the service - will default if not provided
+ * @param [overriddenLoadWithXhr] loadWithXhr function matching the signature of the one provided by Cesium - provided
```
Comment (kring) [inline lib/Models/GNAFApi.js]: `loadWithXhr` has built in support for unit testing by setting `loadWithXhr.load`.  Also you can use `jasmine.Ajax` to override XMLHttpRequest itself.  So I don't think this extra parameter is necessary.

File: lib/Models/GNAFApi.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+/**
+ * Geocodes a plain-text address. Returns results along with their scores.
+ *
+ * If a bounding box is provided, this is queried first - if that query doesn't get a number of results >= to maxResults,
+ * the query will be tried again without the bounding box and these results added to the end of the results from the
+ * initial query. Results that were within the initial bounding box will have {locational: true} set in their object.
+ * Results that are present in both searches will only appear once with locational set to true.
+ *
+ * @param searchTerm The address to geocode
+ * @param [boundingBox] An optional object with the format {topLeft: {latitude, longitude}, bottomRight: {latitude, longitude}}
```
Comment (kring) [inline lib/Models/GNAFApi.js]: It'd be good to use Cesium's `Rectangle` type instead of an object literal.

**Comment by kring** [comment]
Travis broken because of the jsdoc errors.

File: lib/Map/zoomRectangleFromPoint.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+// Server does not return information of a bounding box, just a location.
+// BOUNDING_BOX_SIZE is used to expand a point
+var DEFAULT_BOUNDING_BOX_SIZE = 0.2;
+
+module.exports = function createZoomToFunction(latitude, longitude, boundingBoxSize) {
+    boundingBoxSize = defaultValue(boundingBoxSize, DEFAULT_BOUNDING_BOX_SIZE);
+
+    var south = parseFloat(latitude) + boundingBoxSize / 2;
+    var west = parseFloat(longitude) - boundingBoxSize / 2;
+    var north = parseFloat(latitude) - boundingBoxSize / 2;
```
Comment (kring) [inline lib/Map/zoomRectangleFromPoint.js]: North should be `+` and South should be `-`.

Comment (kring) [inline lib/Map/zoomRectangleFromPoint.js]: Just realized you copied this from `GazetteerSearchProviderViewModel`, but yeah, that code was wrong (probably my fault).

**Comment by kring** [comment]
Looks great other than the few comments!

**Comment by AlexGilleran** [comment]
Hopefully fixed those :)

**Pull request merged**